src/taipy/core/config/data_node_config.py

```diff
@@ -157,49 +163,72 @@ class DataNodeConfig(Section):
             _REQUIRED_READ_FUNCTION_GENERIC_PROPERTY,
             _REQUIRED_WRITE_FUNCTION_GENERIC_PROPERTY,
         ],
-        _STORAGE_TYPE_VALUE_JSON: [_REQUIRED_DEFAULT_PATH_JSON_PROPERTY],
+        _STORAGE_TYPE_VALUE_JSON: [],
         _STORAGE_TYPE_VALUE_PARQUET: [],
     }
 
     _OPTIONAL_PROPERTIES = {
-        _STORAGE_TYPE_VALUE_GENERIC: [
-            _OPTIONAL_READ_FUNCTION_PARAMS_GENERIC_PROPERTY,
-            _OPTIONAL_WRITE_FUNCTION_PARAMS_GENERIC_PROPERTY,
-        ],
-        _STORAGE_TYPE_VALUE_CSV: [
-            _OPTIONAL_EXPOSED_TYPE_CSV_PROPERTY,
-            _OPTIONAL_DEFAULT_PATH_CSV_PROPERTY,
-            _OPTIONAL_HAS_HEADER_CSV_PROPERTY,
-        ],
-        _STORAGE_TYPE_VALUE_EXCEL: [
-            _OPTIONAL_EXPOSED_TYPE_EXCEL_PROPERTY,
-            _OPTIONAL_DEFAULT_PATH_EXCEL_PROPERTY,
-            _OPTIONAL_HAS_HEADER_EXCEL_PROPERTY,
-            _OPTIONAL_SHEET_NAME_EXCEL_PROPERTY,
-        ],
-        _STORAGE_TYPE_VALUE_IN_MEMORY: [_OPTIONAL_DEFAULT_DATA_IN_MEMORY_PROPERTY],
-        _STORAGE_TYPE_VALUE_SQL_TABLE: [_OPTIONAL_EXPOSED_TYPE_SQL_PROPERTY, _OPTIONAL_DB_EXTRA_ARGS_SQL_PROPERTY],
-        _STORAGE_TYPE_VALUE_SQL: [_OPTIONAL_EXPOSED_TYPE_SQL_PROPERTY, _OPTIONAL_DB_EXTRA_ARGS_SQL_PROPERTY],
-        _STORAGE_TYPE_VALUE_MONGO_COLLECTION: [
-            _OPTIONAL_CUSTOM_DOCUMENT_MONGO_PROPERTY,
-            _OPTIONAL_DB_USERNAME_MONGO_PROPERTY,
-            _OPTIONAL_DB_PASSWORD_MONGO_PROPERTY,
-            _OPTIONAL_DB_EXTRA_ARGS_MONGO_PROPERTY,
-        ],
-        _STORAGE_TYPE_VALUE_PICKLE: [_OPTIONAL_DEFAULT_PATH_PICKLE_PROPERTY, _OPTIONAL_DEFAULT_DATA_PICKLE_PROPERTY],
-        _STORAGE_TYPE_VALUE_JSON: [_OPTIONAL_ENCODER_JSON_PROPERTY, _OPTIONAL_DECODER_TYPE_JSON_PROPERTY],
-        _STORAGE_TYPE_VALUE_PARQUET: [
-            _OPTIONAL_EXPOSED_TYPE_PARQUET_PROPERTY,
-            _OPTIONAL_DEFAULT_PATH_PARQUET_PROPERTY,
-            _OPTIONAL_COLUMNS_PARQUET_PROPERTY,
-            _OPTIONAL_COMPRESSION_PARQUET_PROPERTY,
-        ],
+        _STORAGE_TYPE_VALUE_GENERIC: {
+            _OPTIONAL_READ_FUNCTION_PARAMS_GENERIC_PROPERTY: None,
+            _OPTIONAL_WRITE_FUNCTION_PARAMS_GENERIC_PROPERTY: None,
+        },
+        _STORAGE_TYPE_VALUE_CSV: {
+            _OPTIONAL_DEFAULT_PATH_CSV_PROPERTY: None,
+            _OPTIONAL_HAS_HEADER_CSV_PROPERTY: True,
+            _OPTIONAL_EXPOSED_TYPE_CSV_PROPERTY: _DEFAULT_EXPOSED_TYPE,
```

Comment:
should we have a specific _DEFAULT_EXPOSED_TYPE for each datanode, I think we can use a share default exposed type value for now, but when we want to expand many more data node types it will be handy to have a specific one for each

Reply:
That's an interesting question, Indeed.

It depends on what is the future of the `exposed_type` attribute name. Today, it represents the return type of the tabular data nodes' read method. We don't use the `exposed_type` for Json or mongo data nodes. We use default or custom documents.
If we use `exposed_type` only for tabular data nodes, the default exposed type could be the same for all. Or not...
If we extend the attribute name `exposed_type` to other types of data (document-based, graph-based, literal, ...), we most probably want to have a specific default value per storage type.

Let s keep this in mind.